### ISSUE TITLE
Eurocrops Hierarchical Crop and Agriculture Taxonomy Extension

### ISSUE THREAD

**Issue by cholmes** [fiboa/extensions#11]
There's a number of eurocrops datasets that we're converting, and their Hierarchical Crop and Agriculture Taxonomy seems to be emerging as a solid crop classification framework. We should make an extension to be able to validate against it.

A few questions:

 - To prefix or no? The three fields are EC_trans_n, EC_hcat_n and EC_hcat_c. They seem pretty unique already, and indeed EC_ is basically a prefix. Do we need hcat:EC_trans_name? Do we map the prefix to our way? Like ec:trans_name ? 
 - Make more clear? The original field names were designed to fit in the 10 character shapefile limit. We aren't targeting shapefile at all for fiboa (since we have no 10 character naming limitations). So do we try out some more understandable names? Like ec:translated_name, ec:hcat_name, ec:hcat_code? 
 - How should we validate the values? It seems like it'd be ideal to check the code and the name against valid values. But not sure how best to do that. There's almost 400 potential values in [hcat3](https://github.com/maja601/EuroCrops/blob/main/hcat_core/HCAT3.csv) - do we just have a big long enum? And for the code do we just check greater than 3000000000 and less than 3399000001? Or do we try to actually check the values? 
 - Ideally it seems like there's be a validation tool to make sure the code and name align, though that's probably outside the scope of fiboa.
 - Do we just use HCAT3? Or try to also cover hcat 2 and 1? See https://github.com/maja601/EuroCrops/tree/main/hcat_core

See also https://github.com/fiboa/specification/issues/36 for discussion of prefixes.

**Comment by m-mohr** [comment]
They are pretty much prefixed with EC_, I think we can go without and keep them as they are for "backward compatibility".

For dependencies between fields, we could introduce something in fiboa Schema. I feel like that could be a quite common use case.

A long enum is fine, I think.

**Comment by cholmes** [comment]
I worked on this for a bit and got it to a state that's about ready for a 0.1, see https://github.com/fiboa/hcat-extension

I ended up going for 'nicer' looking names. Particularly I wanted to expand to hcat_code and hcat_name, so I ended up doing the ec:hcat_code and ec:hcat_name, and then ec:translated_name. 

But I'm happy to change them, though I'm not sure of the 'value' of backwards compatibility. 

Didn't do the long enum yet. I'll see if I can do that, and then maybe release 0.1 from there. I'll probably try to break the above questions into issues in the extension repo.
